src/subcommands/sign.rs

```diff
@@ -21,32 +23,51 @@ pub struct Sign {
     input_data: String,
 }
 
+#[derive(Serialize, Deserialize)]
+struct EccSignature {
+    r: IntegerAsn1,
+    s: IntegerAsn1,
+}
+
 impl Sign {
     /// Signs data.
     pub fn run(&self, basic_client: BasicClient) -> Result<()> {
-        let alg = key_attributes(&basic_client, &self.key_name)?
+        let alg = basic_client
+            .key_attributes(&self.key_name)?
             .policy
             .permitted_algorithms;
 
-        //TODO: depending on the hash in the algorithm, we need to hash input data
-
         let signature = match alg {
             Algorithm::AsymmetricSignature(alg) => {
+                info!("Hashing data...");
+                let hash = match alg.hash() {
+                    Some(SignHash::Specific(hash)) => hash_data(self.input_data.as_bytes(), hash)?,
+                    _ => {
+                        err!("Asymmetric signing algorithm ({:?}) is not supported", alg);
+                        return Err(ToolErrorKind::NotSupported.into());
+                    }
+                };
                 info!("Signing data...");
-                Ok(basic_client.psa_sign_hash(
-                    self.key_name.clone(),
-                    self.input_data.as_bytes(),
-                    alg,
-                )?)
+                let mut sig = basic_client.psa_sign_hash(self.key_name.clone(), &hash, alg)?;
+                if alg.is_ecc_alg() {
+                    let r = IntegerAsn1::from_bytes_be_unsigned(sig.split_off(sig.len() / 2));
+                    sig = picky_asn1_der::to_vec(&EccSignature {
+                        r,
+                        s: IntegerAsn1::from_bytes_be_unsigned(sig),
+                    })
+                    .unwrap();
```

Comment:
I am ok to keep this for now but I think we should add a new error variant and convert from that

Reply:
Ah, lol, I forgot about that one, will fix in another PR